rfcs/0035-create-connection.md

```diff
@@ -0,0 +1,143 @@
+
+For security related fields (SSL/SASL), the fields are identical to [this doc](https://materialize.com/docs/sql/create-connection/#kafka-ssl).
+
+**Notice**: Risingwave will not check the connection to Kafka is valid or not.
+
+> Some notes for implementation
+>
+> 1. Implement libkafka's DNS resolve callback
+> 2. Store the dns mapping from the user's brokers address to private link endpoint ip
+> 3. Interface for creating and modifying dns mappings
+
+#### Kinesis
+
+A Kinesis connection establishes a link to a Kinesis stream. Users can use Kinesis connections to create Kinesis sources and sinks. The following fields are required to create a Kinesis connection:
+
+```sql
+CREATE CONNECTION demo_connection TO KINESIS
+  (aws.region='user_test_topic',
+   endpoint='172.10.1.1:9090,172.10.1.2:9090',
+   aws.credentials.role.arn='arn:aws-cn:iam::602389639824:role/demo_role',
+   aws.credentials.role.external_id='demo_external_id',
+   ...
+  );
+```
+
+Accepted fields are listed below, which is a subset of [the doc](https://www.risingwave.dev/docs/current/create-source-kinesis/#parameters).
+
+|Field|Required|Description|
+|--|--|--|
+|`aws.region`| yes | AWS service region. For example, US East (N. Virginia). |
+|`endpoint`| Optional |URL of the entry point for the AWS Kinesis service.|
+|`aws.credentials.access_key_id`| Conditional | This field indicates the access key ID of AWS. It must appear in pairs with aws.credentials.secret_access_key.|
+|`aws.credentials.secret_access_key`|Conditional | This field indicates the secret access key of AWS. It must appear in pairs with aws.credentials.access_key_id.|
+|`aws.credentials.session_token`|Optional| The session token associated with the temporary security credentials.|
+|`aws.credentials.role.arn`|Optional| The Amazon Resource Name (ARN) of the role to assume.|
+|`aws.credentials.role.external_id`|Optional|The [external id](https://aws.amazon.com/blogs/security/how-to-use-external-id-when-granting-access-to-your-aws-resources/) used to authorize access to third-party resources.|
+
+**Notice**: Risingwave will not check the connection to Kinesis is valid or not.
+
+#### Pulsar
+
+```sql
+CREATE CONNECTION demo_connection TO PULSAR
+  (service.url = 'pulsar://localhost:6650/',
+  admin.url='http://localhost:8080',);
+```
+
+The accepted fields are listed below, which is a subset of [the doc](https://www.risingwave.dev/docs/current/create-source-pulsar/#parameters).
+
+|Field|Required|Description|
+|--|--|--|
+|`service.url`| Required| Address of the Pulsar service.	|
+|`admin.url`|Required| Address of the Pulsar admin.|
+
+**Notice**: Risingwave will not check the connection to Pulsar is valid or not.
+
+## Future possibilities
+
+* store some sensitive information, eg. passwords and SSL keys, in a encrypted way (use `CREATE SECRET`)
```

Comment:
Question: In Flink if you connect to a catalog e.g. [JDBC database](https://nightlies.apache.org/flink/flink-docs-master/docs/connectors/table/jdbc/#jdbc-catalog), you also automatically get the schema of tables in that database. Will we support this? Does Materialize support this?

Reply:
e.g.

```sql
CREATE CATALOG my_catalog WITH(
    'type' = 'jdbc',
    'default-database' = '...',
    'username' = '...',
    'password' = '...',
    'base-url' = '...'
);

USE CATALOG my_catalog;

SHOW TABLES;
-- Will print a list of tables

DESCRIBE TABLE foo;
-- Will print the schema of that table.
```

Reply:
I don't think it is related to `connection`, which specifies the parameters used when connecting to external systems. 
The feature described above needs further investigation.